compiler/modules/nback/module.bal

```diff
@@ -102,10 +102,17 @@ function orderByFwdTargets(bir:BasicBlock[] blocks, bir:Label label, boolean[] v
     ordered.push(label);
 }
 
+isolated function index(UsedSemType ty) returns int {
+    return ty.index;
+}
+
 function createTypeUsage(table<UsedSemType> usedSemTypes) returns TypeUsage {
     byte[] uses = [];
     t:SemType[] types = [];
-    foreach var used in usedSemTypes {
+    // JBUG 36056 can't directly iterate over usedSemTypes
+    UsedSemType[] usedSemTy = usedSemTypes.toArray();
+    usedSemTy = usedSemTy.sort("ascending", index);
```

Comment:
How about `from var u in usedSemType select u`? That should give them in the right order.

Reply:
fixed (used `toArray()` function since directly using the query expression without it causes the same JBUG)